### PR TITLE
PCVL-479 sampler iterations refresh circuit

### PR DESCRIPTION
**Pull request by ericbrts**

### CODE REVIEW

File: perceval/components/processor.py [hunk 1]
```diff
@@ -54,6 +54,7 @@ class Processor(AProcessor):
         >>> p = Processor("SLOS", BS() // PS() // BS())
 
     :param source: the Source used by the processor (defaults to perfect source)
+    :param name: a textual name for the processor (defaults to "Local processor")
     """
```
Comment (MarionQuandela) [inline perceval/components/processor.py:58]: Do we want to enrich the documentation and be standard to the sphix documentation ?

    """
    Generic definition of processor as a source + components (both unitary and non-unitary) + ports
    + optional post-processing logic

    :param backend: Name or instance of a simulation backend
    :type backend (Union[ABackend, str]):
    :param m_circuit: can either be:

        - an int: number of modes of interest (MOI). A mode of interest is any non-heralded mode.
        >>> p = Processor("SLOS", 5)

        - a circuit: the input circuit to start with. Other components can still be added afterwards with `add()`
        >>> p = Processor("SLOS", BS() // PS() // BS())

        , defaults to None.

    :type m_circuit (Union[int, ACircuit], optional):


    :param source: the Source used by the processor, defaults to perfect source.
    :type source (Source, optional):
    :param name: a textual name for the processor, defaults to "Local processor".
    :type name (str, optional):
    """
    
image

File: perceval/components/processor.py [hunk 2]
```diff
@@ -252,6 +253,9 @@ def probs(self, progress_callback: Callable = None) -> Dict:
         if self._simulator is None:
```
Comment (MarionQuandela) [inline perceval/components/processor.py:253]: Why the commented assert on l252 ?

File: perceval/components/processor.py [hunk 3]
```diff
@@ -252,6 +253,9 @@ def probs(self, progress_callback: Callable = None) -> Dict:
...
             from perceval.simulators import SimulatorFactory  # Avoids a circular import
             self._simulator = SimulatorFactory.build(self)
+        else:
+            self._simulator.set_circuit(self.linear_circuit() if self._is_unitary else self.components)
```
Comment (MarionQuandela) [inline perceval/components/processor.py:257]: Why ?

File: docs/source/notebooks/Graph States and representation.ipynb [hunk 4]
```diff
@@ -72,9 +72,9 @@
     "\n",
     "<br/>\n",
     "\n",
-    "Therefore we can write the action of a CZ gate as: $CZ^{\\{1,2\\}}|0\\rangle_1 |\\pm\\rangle_2 = |0\\rangle_1 |\\pm\\rangle_2$     and     $CZ^{\\{1,2\\}}|0\\rangle_1 |\\pm\\rangle_2 = |1\\rangle_1 |\\mp\\rangle_2$.\n",
+    "Therefore, we can write the action of a CZ gate as: $CZ^{\\{1,2\\}}|0\\rangle_1 |\\pm\\rangle_2 = |0\\rangle_1 |\\pm\\rangle_2$     and     $CZ^{\\{1,2\\}}|1\\rangle_1 |\\pm\\rangle_2 = |1\\rangle_1 |\\mp\\rangle_2$.\n",
     "\n",
-    "Let’s illustrate this with an example. The associated graph of: $|\\Psi_{graph}\\rangle = CZ^{\\{0,1\\}}\\, CZ^{\\{1,2\\}}\\, CZ^{\\{0,2\\}}\\, CZ^{\\{3,2\\}}|+\\rangle_0 |+\\rangle_1 |+\\rangle_2 |+\\rangle_3$ is the following. The graph is displayed later on. We will now see different way to create and represent these states."
+    "Let’s illustrate this with an example. The associated graph of: $|\\Psi_{graph}\\rangle = CZ^{\\{0,1\\}}\\, CZ^{\\{1,2\\}}\\, CZ^{\\{0,2\\}}\\, CZ^{\\{3,2\\}}|+\\rangle_0 |+\\rangle_1 |+\\rangle_2 |+\\rangle_3$ is the following. The graph is shown later on. We will now see several ways to create and display these states."
```
Comment (MarionQuandela) [inline docs/source/notebooks/Graph States and representation.ipynb:77]: The graph is shown/displayed later on --> The graph is generated as an output later on ?

File: tests/test_sampler.py [hunk 5]
```diff
@@ -114,13 +114,18 @@ def test_sampler_iterator():
...
-            {"circuit_params": {"phi1": 0.03}, "input_state": pcvl.BasicState([1, 1]), "min_detected_photons": 1},
-            {"circuit_params": {"phi1": 0.03}, "input_state": pcvl.BasicState([1, 1]), "min_detected_photons": 1},
+            {"circuit_params": {"phi1": 0.5}, "input_state": pcvl.BasicState([1, 1]), "min_detected_photons": 1},
+            {"circuit_params": {"phi1": 0.9}, "input_state": pcvl.BasicState([1, 1]), "min_detected_photons": 1},
             {"circuit_params": {"phi1": 1.57}, "input_state": pcvl.BasicState([1, 0]), "min_detected_photons": 1}
         ]
         sampler.add_iteration_list(iteration_list)
-        res = sampler.probs()
-        assert len(res['results_list']) == len(iteration_list)
+        rl = sampler.probs()['results_list']
```
[the remark this answers is not in the document]
Comment (ericbrts) [reply tests/test_sampler.py:122]: Yes

File: docs/source/notebooks/Graph States and representation.ipynb (hunk 4 above)
Comment (ericbrts) [reply docs/source/notebooks/Graph States and representation.ipynb]: Done

**Pull request merged**